src/util.coffee

```diff
@@ -32,3 +32,25 @@
                                       node?.tag is 'tag:yaml.org,2002:map'
 
 @NON_PRINTABLE = /[^\x09\x0A\x0D\x20-\x7E\x85\xA0-\uD7FF\uE000-\uFFFD]/
+
+@stringifyJSONCircularDepth = (json, depth = 1000) ->
```

Comment:
I wonder when would a cycle ever be broken, why would counting to 500 be more effective than counting to 2?

What we do with includes is keep a list of all the URIs for the include, when a single one in the path is repeated, we abort with a circular reference error.